lib/ex_dtls.ex

```diff
@@ -67,6 +73,16 @@ defmodule ExDTLS do
 
   @doc """
   Starts ExDTLS GenServer process linked to the current process.
+
+  Accepts a keyword list with the following options (`t:opts_t/0`):
+  * `client_mode` - `true` if ExDTLS module should work as a client or `false` if as a server
+  * `dtls_srtp` - `true` if DTLS-SRTP handshake should be performed or `false` if a normal one
+  * `pkey` - private key to use in this SSL context. Must correspond to `cert`
+  * `cert` - certificate to use in this SSL context. Must correspond to `pkey`
+  * `impl` - `:nif` if ExDTLS should execute native code as NIFs or `:cnode` to use a C node.
+    By default C node implementation is used.
+
+  If both `pkey` and `cert` are not passed `ExDTLS` will generate key and certificate on its own.
```

Comment:
Because IMO the description of how the options work should be as close as possible to the function that accepts them. This also provides a better UX for language server users that will get the description on hover. This exactly matches the GenServer docs, where [`start_link`](https://hexdocs.pm/elixir/GenServer.html#start_link/3) has the description, `start` refers to `start_link` and the typespec is used by both, without additional description.

This is quite a common pattern in the docs I saw, for example, found also in https://hexdocs.pm/elixir/Registry.html#start_link/1.

Using `Keyword.t()` as a type would discard type info that `opts_t` provides, so for example a `ExDTLS.start_link(invalid_key: false)` would not generate dialyzer warning